src/main/java/com/redhat/cloud/notifications/routers/UserConfigService.java

```diff
@@ -72,6 +72,10 @@
                     }
                 })));
 
+        /*
+         * (Un)subscriptions will be run sequentially because of the concatenation.
+         * This is mandatory because the Hibernate session must not be used concurrently.
+         */
```

Comment:
IIUC we should never use "merge" when dealing with the database?

Reply:
Merging should be avoided indeed because it runs things concurrently, just like `Uni.combine().all()`. `concatenate` on the other hand runs things sequentially which should be the preferred way to interact with the database.